ratik/src/main/java/org/opencadc/ratik/ArtifactValidator.java

```diff
@@ -160,26 +162,49 @@ protected void validateLocal(Artifact local)
         log.debug("checking explanation 0");
         Artifact remote = getRemoteArtifact(local.getURI());
         if (remote != null) {
+            boolean multipleCopies = false;
+            // if L == storage, get the Artifact count from global
+            if (this.remoteSite == null) {
+                Integer count = this.getRemoteArtifactCount(local.getURI());
+                if (count != null && count > 1) {
+                    multipleCopies = true;
+                }
+            }
             try {
                 log.debug("starting transaction");
                 this.transactionManager.startTransaction();
                 log.debug("start txn: OK");
 
                 this.artifactDAO.lock(local);
-                log.info(String.format("ArtifactValidator.deleteArtifact Artifact.id=%s Artifact.uri=%s reason=local-filter-policy-change",
-                                       local.getID(), local.getURI()));
-                this.artifactDAO.delete(local.getID());
+                if (this.remoteSite != null) {
+                    // if L==global, delete Artifact, do not create a DeletedStorageLocationEvent
+                    log.info(String.format(
+                        "ArtifactValidator.deleteArtifact Artifact.id=%s Artifact.uri=%s reason=local-filter-policy-change",
+                        local.getID(), local.getURI()));
+                    this.artifactDAO.delete(local.getID());
+                } else if (multipleCopies) {
+                    // if L==storage, delete Artifact only if multiple copies in global, and create a DeletedStorageLocationEvent
+                    log.info(String.format(
+                        "ArtifactValidator.deleteArtifact Artifact.id=%s Artifact.uri=%s "
+                            + "reason=local-filter-policy-change, multiple copies in global",
+                        local.getID(), local.getURI()));
+                    this.artifactDAO.delete(local.getID());
 
-                if (this.remoteSite == null) {
-                    DeletedStorageLocationEvent deletedStorageLocationEvent =
-                        new DeletedStorageLocationEvent(local.getID());
-                    log.info(String.format("ArtifactValidator.createDeletedStorageLocationEvent event=%s reason=local-filter-policy-change",
-                                           deletedStorageLocationEvent));
+                    DeletedStorageLocationEvent deletedStorageLocationEvent = new DeletedStorageLocationEvent(local.getID());
+                    log.info(String.format(
+                        "ArtifactValidator.createDeletedStorageLocationEvent event=%s reason=local-filter-policy-change",
+                        deletedStorageLocationEvent));
                     this.deletedStorageLocationEventDAO.put(deletedStorageLocationEvent);
+                } else {
+                    log.info(String.format(
+                        "ArtifactValidator.didNotDeleteArtifact Artifact.id=%s Artifact.uri=%s "
```

Comment:
if you look in tantar (InventoryIsAlwaysRight), there is a "no-op" event type there that is comparabel to the situation here. ".delayAction" that seems to be roughly the same meaning as here: it's too soon to do this action, so leaving this discrepancy alone for now.

the value of each k=v in the log message needs to be a single string (no punctuation or spaces) in order to be searchable. Here, you just need `reason=local-filter-policy-change` without the "but".

Reply:
Updated the log message.

Reply:
The builds for cadc-storage-adapter-ad and critwall were failing on the new RangeNotSatisfiableException in the cadc-util net classes. Included the fixes with this PR.